### ISSUE TITLE
tsconfig.json values for target and module being changed to incompatible numeric values

### ISSUE THREAD

**Issue by jamesbrobb** [ivogabe/gulp-typescript#471]
I'm attempting to upgrade an existing project from TS 1.8 to TS 2, but have come across an issue.

**Versions**

npm 3.10.9
typescript 2.1.4
gulp-typescript 3.1.3

**Expected behavior:**

The supplied tsconfig should not cause any errors

**Actual behavior:**

```
error TS5024: Compiler option 'target' requires a value of type string.
error TS5024: Compiler option 'module' requires a value of type string.
error TS5023: Unknown compiler option 'suppressOutputPathCheck'.
```

`tsProject.options` returns the following, with the numeric values that are set for `target` and `module` appearing to cause the problem.

```
{ 
  target: 1,
  module: 4,
  emitDecoratorMetadata: true,
  experimentalDecorators: true,
  removeComments: false,
  noImplicitAny: false,
  typeRoots: [ 'node_modules/@types' ],
  types: 
   [ 'angular',
     'angular-animate',
     'angular-cookies',
     'angular-mocks',
     'angular-sanitize',
     'angular-ui-router',
     'autobahn',
     'es6-promise',
     'gsap',
     'isomorphic-fetch',
     'jasmine',
     'lodash',
     'raven-js' ],
  sourceMap: true,
  suppressOutputPathCheck: true
}
```

That said, on my existing project, which uses `typescript 1.8.10` and `gulp-typescript 2.13.0` the output is exactly the same, but with no problems.

If i change line 73 of `main.js` to the following, the errors are removed:

```javascript
//result.options.suppressOutputPathCheck = true;
result.options.target = "es5";
result.options.module = "system";
```

I also notice that, even though i only call `ts.createProject()` once, internally, it's getting called 3 times.

**gulpfile:**

```javascript
var gulp = require('gulp'),
      ts = require('gulp-typescript'),
      sourcemaps = require('gulp-sourcemaps');

var tsProject = ts.createProject('tsconfig.json');

gulp.task('scripts', function() {

    var result = tsProject.src()
        .pipe(sourcemaps.init())
        .pipe(tsProject());

    return result.js
        .pipe(sourcemaps.write())
        .pipe(gulp.dest('dist'));
});
```

**tsconfig.json**

```json
{
    "compilerOptions": {
        "target": "es5",
        "module": "system",
        "sourceMap": false,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "removeComments": false,
        "noImplicitAny": false,
        "typeRoots": [
            "node_modules/@types"
        ],
        "types": [
            "angular",
            "angular-animate",
            "angular-cookies",
            "angular-mocks",
            "angular-sanitize",
            "angular-ui-router",
            "autobahn",
            "es6-promise",
            "gsap",
            "isomorphic-fetch",
            "jasmine",
            "lodash",
            "raven-js"
        ]
    },
    "include": [
        "src/**/*.ts"
    ]
}
```

**Comment by ivogabe** [comment]
I cannot reproduce this. Given that you see `Unknown compiler option 'suppressOutputPathCheck'.`, you might be using an older version of TypeScript, since that option was added in TS 2.0. Note that gulp-typescript uses the local installation of TypeScript, not the global one (so you need to install typescript using npm *without* the `-g` flag). If you do have the correct version of TypeScript installed, can you then create a small but complete example that reproduces your issue and upload it here as a .zip?

**Comment by jamesbrobb** [comment]
Does `gulp-typescript` 2.13.0 work with TS 2.0 then? I'm using it on another project that uses TS 1.8, but notice that it also forces the same settings in `main.js`?

```
tsSettings.sourceMap = true;
// Suppress errors when providing `allowJs` without `outDir`.
tsSettings.suppressOutputPathCheck = true;
```

**Comment by ivogabe** [comment]
No, you have to use gulp-typescript 3.x with typescript 2.x, or gulp-typescript 2.x with typescript 1.x.

**Comment by ivogabe** [comment]
You might need to install the latest TypeScript version locally, using `npm install typescript --save-dev`.

**Comment by samih-dev** [comment]
thanks @ivogabe installing typescrpt locally solved the issue for me!
